lib/Data/Chronicle.pm

```diff
@@ -0,0 +1,89 @@
+package Data::Chronicle;
+
+=head1 NAME
+
+Data::Chronicle - Chronicle storage system
+
+=head1 VERSION
+
+Version 0.06
+
+=cut
+
+our $VERSION = '0.06';
+
+=head1 DESCRIPTION
+
+This package contains two modules (Reader and Writer) which can be used to store and retrieve information
+on an efficient storage with below properties:
+
+=head2 Timeliness
+
+It is assumed that data to be stored are time-based meaning they change over time and the latest version is most important for us.
+
+=head2 Efficient
+
+The module uses Redis cache to provide efficient data storage and retrieval.
+
+=head2 Persistent
+
+In addition to caching every incoming data, it is also stored in PostgresSQL for future retrieval.
+
+=head2 Transparent
+
+This modules hides all the details about caching, database structure and ... from developer. He only needs to call a method
+to save data and another method to retrieve it. All the underlying complexities are handled by the module.
+
+Note that you will need to pass `cache_writer`, `cache_reader` and `db_handle` to the `Data::Chronicle::Reader/Writer` modules. These three arguments, provide access to your Redis and PostgreSQL which will be used by Chronicle modules.
+
+`cache_writer` and `cache_reader` should be to be able to get/set given data under given key (both of type string). `db_handle` should be capable to store and retrieve data with `category`,`name` in addition to the timestamp of data insertion. So it should be able to retrieve data for a specific timestamp, category and name. Category, name and data are all string. This can easily be achieved by defining a table in you database containing these columns: `timestamp, category, name, value`. 
+
+=head1 METHODS
+
+There are four important methods this module provides:
+
+=head2 set (in L<Data::Chronicle::Writer>)
```

Comment:
Minor style note: if you define the heading in the other file as

```perl
=head2 set
```

then you can use `L<Data::Chronicle::Writer/set>` to provide a direct link.

Normally I'd recommend specifying the headings this way - just the method name - with the example usage in the text under the heading:

```
=head2 set

Set a value.

    $writer->set("category1", "name1", $value1)
    $writer->set("category1", "name1", $value1, Date::Utility->new("2016-08-01 00:06:00"))

=cut
```

but again, that's a style issue - not all CPAN authors do this.